test/android/core/javatests/src/org/xwalk/core/xwview/test/SettingsTest.java

```diff
@@ -17,10 +19,80 @@
     private static final String USER_AGENT =
             "Chrome/44.0.2403.81 Crosswalk/15.44.376.0 Mobile Safari/537.36";
     private static final String LANGUAGE = "jp";
+    private TestWebServer mWebServer;
 
     @Override
     public void setUp() throws Exception {
         super.setUp();
+        mWebServer = TestWebServer.start();
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        mWebServer.shutdown();
+        super.tearDown();
+    }
+
+    @MediumTest
+    @Feature({"Settings"})
+    public void testCacheMode() throws Throwable {
+        final String htmlPath = "/testCacheMode.html";
+        final String url = mWebServer.setResponse(htmlPath, "response", null);
+        final String htmlNotInCachePath = "/testCacheMode-not-in-cache.html";
+        final String urlNotInCache = mWebServer.setResponse(htmlNotInCachePath, "", null);
+
+        clearCacheOnUiThread(true);
+        assertEquals(XWalkSettings.LOAD_DEFAULT, getCacheMode());
+
+        setCacheMode(XWalkSettings.LOAD_CACHE_ELSE_NETWORK);
+        loadUrlSync(url);
+        assertEquals(1, mWebServer.getRequestCount(htmlPath));
```

Comment:
This number is depends on how many request to TestWebServer.
For example, clear all the previous cache by `clearCacheOnUiThread`, then set the cache mode to LOAD_CACHE_ELSE_NETWORK, which means `use cached resources when they are available, even if they have expired. Otherwise load resources from the network.`, and then after the loadUrlSync, **one time request** to web server is needed because **no cache currently**.  And then, **the page is cached**, so after the 2nd loadUrlSync, as the previous **cache mode setting**, there is **no need to request** webserver, so the mWebServer.getRequestCount should also return 1, just as the same before 2nd loadUrlSync.
Others and so on..